File: FE/src/app/dto/InputInvoiceDto.ts
import {ProductInputDto} from './ProductInputDto';
import {Supplier} from '../model/supplier';

export class InputInvoiceDto {
  productInputDto: ProductInputDto[];
  supplier: Supplier;


  constructor(productInputDto: ProductInputDto[], supplier: Supplier) {
    this.productInputDto = productInputDto;
    this.supplier = supplier;
  }
}

File: BE/src/main/java/com/example/a0922i1projectmobilephone/dto/input_invoice/InputInvoiceDto.java
package com.example.a0922i1projectmobilephone.dto.input_invoice;

public class InputInvoiceDto {

   private ProductInputDto[] productInputDto;
    private SupplierIdDto supplier;

    public InputInvoiceDto(ProductInputDto[] productInputDto, SupplierIdDto supplier) {
        this.productInputDto = productInputDto;
        this.supplier = supplier;
    }

    public InputInvoiceDto() {
    }

    public ProductInputDto[] getProductInputDto() {
        return productInputDto;
    }

    public void setProductInputDto(ProductInputDto[] productInputDto) {
        this.productInputDto = productInputDto;
    }

    public SupplierIdDto getSupplier() {
        return supplier;
    }

    public void setSupplier(SupplierIdDto supplier) {
        this.supplier = supplier;
    }
}
